lib/router/routes.dart

```diff
@@ -150,19 +106,15 @@ class MyRouter {
 
     // redirect to the login page if the user is not logged in
     redirect: (state) {
-      if ((state.location == loginRoute ||
-              state.location == createAccountRoute) &&
-          !loginState.loggedIn) {
-        return null;
-      }
-      if (!loginState.loggedIn) {
-        return loginRoute;
-      }
-      if ((state.location == loginRoute ||
-              state.location == createAccountRoute) &&
-          loginState.loggedIn) {
-        return homeRoute;
-      }
+      final loginLoc = state.namedLocation(loginRouteName);
+      final loggingIn = state.subloc == loginLoc;
+      final createAccountLoc = state.namedLocation(createAccountRouteName);
+      final creatingAccount = state.subloc == createAccountLoc;
+      final loggedIn = loginState.loggedIn;
+      final homeLoc = state.namedLocation(homeRouteName);
+
+      if (!loggedIn && !loggingIn && !creatingAccount) return loginLoc;
```

Comment:
Would this be better?
if (!loggedIn && !loggingIn || !creatingAccount) return loginLoc;

Reply:
no. precedence means you have to group the two values that are being or'd together otherwise it becomes

```dart
if ((!loggedIn && !loggingIn) || !creatingAccount) return loginLoc;
```